src/plugins/telemetry/public/plugin.test.ts

```diff
@@ -0,0 +1,43 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+import { TelemetryPlugin } from './plugin';
+import { coreMock } from '../../../core/public/mocks';
+import { homePluginMock } from '../../../../src/plugins/home/public/mocks';
+import { screenshotModePluginMock } from '../../../../src/plugins/screenshot_mode/public/mocks';
+import { cloudMock } from '../../../../x-pack/plugins/cloud/public/mocks';
+
+const mockInitializerContext = coreMock.createPluginInitializerContext();
+const home = homePluginMock.createSetupContract();
+const screenshotMode = screenshotModePluginMock.createSetupContract();
+const cloud = cloudMock.createSetup();
+
+describe('TelemetryPublicPlugin', () => {
+  beforeEach(() => jest.clearAllMocks());
```

Comment:
We usually try to avoid using `clearAllMocks` to avoid cleaning potentially pre-feeded mocks from outside (and overall to properly reconstruct the test's sanbox between every test) 

```ts
describe('stuff', () => {
   let mockInitializerContext: ReturnType<typeof coreMock.createPluginInitializerContext>;
   let home: ReturnType<typeof homePluginMock.createSetupContract>;
   let screenshotMode: ReturnType<typeof screenshotModePluginMock.createSetupContract>;
   let cloud: ReturnType<typeof cloudMock.createSetup>;

   beforeEach(() => {
      mockInitializerContext = coreMock.createPluginInitializerContext();
      home = homePluginMock.createSetupContract();
      screenshotMode = screenshotModePluginMock.createSetupContract();
      cloud = cloudMock.createSetup(); 
   });

   it('do some stuff', () => { ... })

})

```